### PR TITLE
GUI: translation extension, refactoring and some improvements

### PR DESCRIPTION
**Pull request by bookmist**
* lang: more verbose error on wrong lang.xml
* gui: refactoring: replace magic strings to enum types, move static vars inside corresponding functions, fix datatypes
* gui: refactoring: minor fixes: global vars as static, add or remove ref if needed
* gui: fix a bug when one button could be mapped to two commands (for screenshots and texture replacement)
* gui: refactoring: Extract functions to draw text centered
* gui: Settings dialog: more translatable strings

### CODE REVIEW

**Review by Zangetsu38 — APPROVED** [review]

File: lang/system/en.xml [hunk 1]
```diff
@@ -604,6 +604,12 @@ Check vita3k.log to see console output for details.
 			<no_modules>No modules present.
 Please download and install the last PS Vita firmware.</no_modules>
 			<refresh_list>Refresh List</refresh_list>
+            <automatic>Automatic</automatic>
+            <automatic_description>Select Automatic mode to use a preset list of modules.</automatic_description>
+            <auto_manual>Auto & Manual</auto_manual>
+            <auto_manual_description>Select this mode to load Automatic module and selected modules from the list below.</auto_manual_description>
+            <manual>Manual</manual>
+            <manual_description>Select Manual mode to load selected modules from the list below.</manual_description>
```
Comment (Zangetsu38) [inline lang/system/en.xml:612]: juste fix decalage here

**Review by Zangetsu38 — APPROVED** [review]
lgtm, good works

File: vita3k/gui/src/controls_dialog.cpp [hunk 2]
```diff
@@ -44,40 +44,31 @@ static constexpr std::array<const char *, 256> SDL_key_to_string{ "[unset]", "[u
...
-    map[26] = emuenv.cfg.keyboard_gui_fullscreen;
-    map[27] = emuenv.cfg.keyboard_gui_toggle_touch;
+    size_t i = 0;
+#define ADD_KEYBOARD_MEMBERS(option_type, option_name, option_default, member_name) \
+    if constexpr (std::string_view(#member_name).starts_with("keyboard_")) {        \
+        map[i++] = int_or_zero(emuenv.cfg.member_name);                             \
+    }
+
+    CONFIG_INDIVIDUAL(ADD_KEYBOARD_MEMBERS)
+#undef ADD_KEYBOARD_MEMBERS
```
Comment (Macdu) [inline vita3k/gui/src/controls_dialog.cpp:68]: That's a bit hacky (although it looks a lot nicer), you should add a comment in config.h saying that all variables starting with "keyboard_" will be considered as key input or something similar.

File: vita3k/gui/src/home_screen.cpp [hunk 3]
```diff
@@ -523,10 +523,11 @@ void select_app(GuiState &gui, const std::string &title_id) {
         LOG_ERROR("App with title id {} not found", title_id);
 }
 
-static constexpr ImU32 ARROW_COLOR = 0xFFFFFFFF; // White
-static float scroll_type, current_scroll_pos, max_scroll_pos;
-
 void draw_home_screen(GuiState &gui, EmuEnvState &emuenv) {
+    static constexpr ImU32 ARROW_COLOR = 0xFFFFFFFF; // White
```
Comment (Macdu) [inline vita3k/gui/src/home_screen.cpp:527]:
```suggestion
    constexpr ImU32 ARROW_COLOR = 0xFFFFFFFF; // White
```
No need to make a constexpr integer static

File: vita3k/gui/src/home_screen.cpp [hunk 4]
```diff
@@ -523,10 +523,11 @@ void select_app(GuiState &gui, const std::string &title_id) {
...
+    static int scroll_type;
+    static float current_scroll_pos, max_scroll_pos;
```
Comment (Macdu) [inline vita3k/gui/src/home_screen.cpp:529]: Looking at this functions, these two variables (current_scroll_pos, max_scroll_pos) are always set before being used, is there a reason they are static?

File: vita3k/gui/src/license_install_dialog.cpp [hunk 5]
```diff
@@ -39,6 +35,20 @@ void draw_license_install_dialog(GuiState &gui, EmuEnvState &emuenv) {
...
+        ZRIF,
+        SUCCESS,
+        FAIL
+    };
+
+    static State state = State::UNDEFINED;
+
+    static std::string title, zRIF;
+    static std::filesystem::path license_path = "";
+    static bool delete_license_file;
```
Comment (Macdu) [inline vita3k/gui/src/license_install_dialog.cpp:50]: From its use, there is no reason to make delete_license_file a static

File: vita3k/gui/src/pkg_install_dialog.cpp [hunk 6]
```diff
@@ -30,19 +30,29 @@
...
     static std::atomic<float> progress(0);
     static std::mutex install_mutex;
     static const auto progress_callback = [&](float updated_progress) {
         progress = updated_progress;
     };
+    static std::filesystem::path pkg_path = "";
+    static std::filesystem::path license_path = "";
+    static std::string title, zRIF;
+    static bool draw_file_dialog = true;
+    static bool delete_pkg_file, delete_license_file;
```
Comment (Macdu) [inline vita3k/gui/src/pkg_install_dialog.cpp:44]: Same, it looks like delete_pkg_file and delete_license_file have no need to be static.

File: vita3k/gui/src/settings_dialog.cpp [hunk 7]
```diff
@@ -485,13 +485,11 @@ void draw_settings_dialog(GuiState &gui, EmuEnvState &emuenv) {
     ImGui::PushStyleColor(ImGuiCol_Text, GUI_COLOR_TEXT_MENUBAR);
     ImGui::SetNextWindowPos(ImVec2(emuenv.viewport_pos.x + (display_size.x / 2.f), emuenv.viewport_pos.y + (display_size.y / 2.f)), ImGuiCond_Always, ImVec2(0.5f, 0.48f));
     const auto is_custom_config = gui.configuration_menu.custom_settings_dialog;
-    auto &settings_dialog = is_custom_config ? gui.configuration_menu.custom_settings_dialog : gui.configuration_menu.settings_dialog;
+    auto settings_dialog = is_custom_config ? gui.configuration_menu.custom_settings_dialog : gui.configuration_menu.settings_dialog;
```
Comment (Macdu) [inline vita3k/gui/src/settings_dialog.cpp:488]:
```suggestion
    auto show_settings_dialog = is_custom_config ? gui.configuration_menu.custom_settings_dialog : gui.configuration_menu.settings_dialog;
```
Can help to understand that it is a boolean

File: vita3k/gui/src/trophy_collection.cpp [hunk 8]
```diff
@@ -123,7 +123,12 @@ static bool load_trophy_progress(IOState &io, const SceUID &progress_input_file,
     return true;
 }
 
-static std::string np_com_id_sort;
+enum class np_com_id_sort_type {
```
Comment (Macdu) [inline vita3k/gui/src/trophy_collection.cpp:126]: Please use pascal case enum names.

File: vita3k/gui/src/home_screen.cpp (hunk 4 above)
Comment (Zangetsu38) [reply vita3k/gui/src/home_screen.cpp:529]: yes, in version on this pr no need static

File: vita3k/gui/src/license_install_dialog.cpp (hunk 5 above)
Comment (Zangetsu38) [reply vita3k/gui/src/license_install_dialog.cpp:50]: is neded here i thinks, if not when you check box is disable directly ? 🤔

Comment (bookmist) [reply vita3k/gui/src/license_install_dialog.cpp:50]: if `delete_license_file` is not static then checkbox does not save its state between frames, so it's impossible to set checkbox on.

Comment (Zangetsu38) [reply vita3k/gui/src/license_install_dialog.cpp:50]: Yes is what i said

**Comment by bookmist** [comment]
@Macdu Ready for review. All fixed, except `bool delete_*_file`

**Review by Macdu — APPROVED** [review]

**Pull request merged**